reascripts/ReaSpeech/tests/TestToolWindow.lua

```diff
@@ -0,0 +1,246 @@
+package.path = '../common/libs/?.lua;../common/vendor/?.lua;' .. package.path
+
+local lu = require('luaunit')
+
+require('mock_reaper')
+require('ImGuiTheme')
+require('Polo')
+require('libs/ToolWindow')
+require('source/Logging')
+require('source/include/globals')
+
+--
+
+TestToolWindow = {
+  _test_class = function()
+    return Polo {
+      init = function(self)
+        ToolWindow.init(self, {
+          title = 'Test Tool Window',
+          width = 300,
+          height = 200,
+          window_flags = 21,
+        })
+      end
+    }
+  end
+}
+
+function TestToolWindow:setUp()
+end
+
+function TestToolWindow:testInit()
+  local test_class = self._test_class()
+
+  local o = test_class.new()
+  lu.assertNotNil(o._tool_window)
+  lu.assertEquals(type(o._tool_window), 'table')
+  lu.assertEquals(o._tool_window.title, 'Test Tool Window')
+  lu.assertEquals(o._tool_window.window_flags, 21)
+  lu.assertEquals(o._tool_window.width, 300)
+  lu.assertEquals(o._tool_window.height, 200)
+  lu.assertEquals(type(o:is_open()), 'boolean')
+  lu.assertEquals(type(o.open), 'function')
+  lu.assertEquals(type(o.close), 'function')
+  lu.assertEquals(type(o.render_separator), 'function')
+  lu.assertEquals(type(o.render), 'function')
+end
+
+function TestToolWindow:testModal()
+  local test_class = Polo {
+    init = function(self)
+      ToolWindow.modal(self, {
+        title = 'Test Tool Window',
+        width = 300,
+        height = 200,
+        window_flags = 21,
+      })
+    end
+  }
+
+  ImGui = ImGui or {}
+  ImGui.WindowFlags_AlwaysAutoResize = function() return 0 end
+  ImGui.WindowFlags_NoCollapse = function() return 1 end
+  ImGui.WindowFlags_NoDocking = function() return 2 end
```

Comment:
Do we need to do this, if mock_reaper also does it? (comment applies to repeated instances)

Reply:
i guess it depends on your definition of "this" 🤔

definitely need the mock definitions. maybe don't need them to be different number values? 🤔🤔